lib/concurrent-ruby/concurrent/utility/processor_counter.rb

```diff
@@ -21,6 +22,23 @@ def physical_processor_count
         @physical_processor_count.value
       end
 
+      def available_processor_count
+        cpu_count = processor_count.to_f
+        quota = cpu_quota
+
+        return cpu_count if quota.nil?
+
+        if quota > cpu_count
```

Comment:
Can quota be higher than processor_count?
Because if not it could be just `cpu_quota || processor_count.to_f` which is quite a bit more readable.

Reply:
Yes it can:

```
$ podman run -it ruby:3.3 bash
root@c7a0b7656940:/# ruby -r etc -e 'p Etc.nprocessors'
4
-- exit
$ podman run --cpus 42 -it ruby:3.3 bash
root@221a93a8b707:/# cat /sys/fs/cgroup/cpu.max
4200000 100000
```

Reply:
Thanks, could you add a comment about it in the code?
(and gah this CPU quota feature seems pretty hacky, doesn't integrate into `sysconf(_SC_NPROCESSORS_ONLN)`, no proper limits, etc.)